src/System.Runtime/tests/System/Int32Tests.cs

```diff
@@ -151,130 +151,238 @@ public static void ToString_InvalidFormat_ThrowsFormatException()
         public static IEnumerable<object[]> Parse_Valid_TestData()
         {
             NumberStyles defaultStyle = NumberStyles.Integer;
-            NumberFormatInfo emptyFormat = new NumberFormatInfo();
-
-            NumberFormatInfo customFormat = new NumberFormatInfo();
-            customFormat.CurrencySymbol = "$";
-
-            yield return new object[] { "-2147483648", defaultStyle, null, -2147483648 };
```

Comment:
> we know that NumberStyles.Integer is a composite type, so the tests are more likely to be more robust if we replace NumberStyles.Integer with the thing we're actually testing

That's fine.  But it also means that tests that were expecting Integer now won't run, e.g.
https://github.com/dotnet/corefx/pull/8085/files#diff-e57db10ee4f89545051b491913e641eaR239

Should https://github.com/dotnet/corefx/pull/8085/files#diff-e57db10ee4f89545051b491913e641eaR239 be changed to instead be something like:
```C#
if ((style & ~NumberStyles.Integer) == 0)
{
    ...
}
```
so that those additional tests get run for styles that are subsets of Integer?